pkg/config/validation.go

```diff
@@ -174,3 +176,52 @@ func validatePodIntegrationOptions(c *configapi.Configuration) field.ErrorList {
 
 	return allErrs
 }
+
+var (
+	validStrategySets = [][]configapi.PreemptionStrategy{
+		{
+			configapi.LessThanOrEqualToFinalShare,
+		},
+		{
+			configapi.LessThanInitialShare,
+		},
+		{
+			configapi.LessThanOrEqualToFinalShare,
+			configapi.LessThanInitialShare,
+		},
+	}
+
+	validStrategySetsStr = func() []string {
+		var ss []string
+		for _, s := range validStrategySets {
+			// Casting because strings.Join requires a slice of strings
+			strategies := *(*[]string)(unsafe.Pointer(&s))
+			ss = append(ss, strings.Join(strategies, ","))
+		}
+		return ss
+	}()
+)
+
+func validateFairSharing(c *configapi.Configuration) field.ErrorList {
+	fs := c.FairSharing
+	if fs == nil {
+		return nil
+	}
+	var allErrs field.ErrorList
+	if !fs.Enable && len(fs.PreemptionStrategies) != 0 {
+		allErrs = append(allErrs, field.Invalid(fsPreemptionStrategiesPath, fs.PreemptionStrategies, "Must be empty when fair sharing is disabled"))
+	}
```

Comment:
Do we need it? I guess it can be handy to keep th configuration when temporarily disabling the feature. 
Similarly as for waitForPodsReady.

Reply:
If there are no drawbacks (similar to additional computation, and potentially bugs in preemptor), I would prefer to apply the @mimowo proposal here.

Reply:
SG. Done